array-manipulation-3/src/zip.js

```diff
@@ -0,0 +1,17 @@
+/* exported zip */
+
+function zip(first, second) {
+  var returnArray = [];
+  if (first.length <= second.length) {
+    for (var i = 0; i < first.length; i++) {
+      var subArray1 = [first[i], second[i]];
+      returnArray.push(subArray1);
+    }
+  } else if (second.length < first.length) {
+    for (var j = 0; j < second.length; j++) {
+      var subArray2 = [first[j], second[j]];
+      returnArray.push(subArray2);
+    }
+  }
```

Comment:
You could simplify this a bit by deciding which array is shorter and then writing the loop once.

```js
var length = Math.min(first.length, second.length);
```